test/module/shared_model/backend_proto/proto_batch_test.cpp

```diff
@@ -65,8 +65,12 @@ TEST(TransactionBatchTest, CreateTransactionBatchWhenValid) {
   auto transaction_batch =
       interface::TransactionBatchFactory::createTransactionBatch(
           txs, validation::DefaultUnsignedTransactionsValidator());
-  ASSERT_TRUE(framework::expected::val(transaction_batch))
-      << framework::expected::err(transaction_batch).value().error;
+  transaction_batch.match(
```

Comment:
Please fix `val` and `err` to accept non-copyable argument and reuse the methods. Same for changes below.